src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/KeyworkerStatsBatchService.kt

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.digital.hmpps.keyworker.services
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.retry.RetryCallback
+import org.springframework.retry.support.RetryTemplate
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.keyworker.model.PrisonKeyWorkerStatistic
+
+@Service
+class KeyworkerStatsBatchService(
+  private val keyworkerStatsService: KeyworkerStatsService,
+  private val prisonSupportedService: PrisonSupportedService,
+  private val defaultRetryTemplate: RetryTemplate
+) {
+  val log: Logger = LoggerFactory.getLogger(this.javaClass.name)
+
+  fun generatePrisonStats() {
+    try {
+      val prisonsWithId = prisonSupportedService.migratedPrisons
+      log.info("There are {} migrated prisons", prisonsWithId.size)
+      for (prison in prisonsWithId.stream()) {
```

Comment:
Don't change now but might be more consistent to use, if possible:`
    prisonsWithId.forEach {
        generatePrisonStatsForPrison(it.prisonId)
    }
}`

Reply:
That's weird, because this is the one bit of code that IntelliJ corrected for me?!